src/CachePeers.cc

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+
+#include "CachePeers.h"
+#include "SquidConfig.h"
+
+CachePeers::Storage::const_iterator
+CachePeers::nextPeerToPing()
+{
+    Assure(size());
+    const auto pos = peersPinged_ % size();
+    ++peersPinged_;
+    return storage.begin() + pos;
+}
+
+void
+CachePeers::remove(CachePeer *p)
+{
+    for (auto it = storage.begin(); it != storage.end(); ++it) {
+        if (it->get() == p) {
+            storage.erase(it);
+            break;
+        }
+    }
+}
+
+const CachePeers &
+CurrentCachePeers()
+{
+    if (!Config.cachePeers) {
+        static CachePeers peers;
+        return peers;
+    }
+    return *Config.cachePeers;
+}
+
+void
+NeighborRemove(CachePeer *peer)
+{
+    Assure(Config.cachePeers);
+    Config.cachePeers->remove(peer);
```

Comment:
This assertion seems inconsistent with ignoring a failure to find the peer in remove(). The official code does not guarantee removal AFAICT. If we want to preserve that behavior in this PR:

```suggestion
    if (Config.cachePeers)
        Config.cachePeers->removeIfFound(peer);
```

If we want to change that behavior, then this assertion is correct, but we need to change the remove() method to assert if the peer is not found.